xapianIndexer.cpp

```diff
@@ -52,6 +52,10 @@ XapianIndexer::XapianIndexer(const std::string& language, const bool verbose) :
 }
 
 XapianIndexer::~XapianIndexer(){
+
+  /* Need to release directory handle first */
+  this->writableDatabase = Xapian::WritableDatabase();
```

Comment:
Is `this->writableDatabase->close()` would have been better ?
It is a bit surprising to create a new database in a destructor.